diff -ur kdelibs-orig/kdeprint/cups/cupsdconf2/cupsdconf.cpp kdelibs-3.5.9/kdeprint/cups/cupsdconf2/cupsdconf.cpp
--- kdelibs-orig/kdeprint/cups/cupsdconf2/cupsdconf.cpp	2005-10-10 17:06:30.000000000 +0200
+++ kdelibs-3.5.9/kdeprint/cups/cupsdconf2/cupsdconf.cpp	2008-02-26 15:03:47.000000000 +0100
@@ -71,10 +71,10 @@
 	//fontpath_ << (datadir_+"/fonts");
 	requestdir_ = findDir(QStringList("/var/spool/cups")
 			<< "/var/cups");
-	serverbin_ = findDir(QStringList("/usr/lib" KDELIBSUFF "/cups")
-			<< "/usr/local/lib" KDELIBSUFF "/cups"
-			<< "/opt/lib" KDELIBSUFF "/cups"
-			<< "/opt/local/lib" KDELIBSUFF "/cups");
+	serverbin_ = findDir(QStringList("/usr/lib/cups")
+			<< "/usr/local/lib/cups"
+			<< "/opt/lib/cups"
+			<< "/opt/local/lib/cups");
 	serverfiles_ = findDir(QStringList("/etc/cups")
 			<< "/usr/local/etc/cups");
 	tmpfiles_ = requestdir_+"/tmp";
