--- kdelibs-3.5.4/kdeprint/cups/kmcupsmanager.cpp.orig	2006-10-01 21:55:49.000000000 +0200
+++ kdelibs-3.5.4/kdeprint/cups/kmcupsmanager.cpp	2006-10-01 21:56:08.000000000 +0200
@@ -85,7 +85,7 @@
 	// change LANG variable so that CUPS is always using
 	// english language: translation may only come from the PPD
 	// itself, or from KDE.
-	setenv("LANG", "en", 1);
+	setenv("LANG", "en_US.UTF-8", 1);
 }
 
 KMCupsManager::~KMCupsManager()
